File: my_py_pkg/number_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64
from example_interfaces.srv import SetBool

class NumberCounterNode(Node):
    def __init__(self):
        super().__init__('number_counter')
        self.counter = 0
        self.number_count_pub = self.create_publisher(Int64, 'number_count', 10)
        self.number_subscriber = self.create_subscription(Int64, 'number', self.callback_number, 10)
        self.reset_counter_srv = self.create_service(SetBool, 'reset_counter', self. callback_reset_counter )
        self.get_logger().info('Number counter has been started.')



    def  callback_number(self, msg):
        self.counter += msg.data 
        #self.get_logger().info(str(self.counter))
        new_msg= Int64()
        new_msg.data = self.counter
        self.number_count_pub.publish(new_msg)        
    
    def callback_reset_counter(self, request: SetBool.Request, response: SetBool.Response):
        if request.data == True:
            self.counter = 0
            response.success = True
            response.message = 'counter has been reset'

        else:
            response.success = False
            response.message = 'Counter not reset'
        
        return response

def main(args=None):
    rclpy.init(args=args)
    node =NumberCounterNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__=='__main__':
    main()

File: my_cpp_pkg/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::chrono_literals;

class NumberCounterNode : public rclcpp::Node
{
public :
    NumberCounterNode() : Node("number_counter"), count_(0)
    {
        subscriber_ = this->create_subscription<example_interfaces::msg::Int64>("number", 10, std::bind(&NumberCounterNode::callbackCounter,
        this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), " Number Counter has been started");
        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number_count", 10);
    }



private :

    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscriber_ ;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;


    void callbackCounter(const example_interfaces::msg::Int64::SharedPtr msg)
    {
       
        count_ = count_ + msg->data;
       
       //RCLCPP_INFO(this->get_logger(), "Current count: %d", count_);
       auto message = example_interfaces::msg::Int64();
       message.data = count_;
       publisher_ ->publish(message);
        
    }
    int  count_;
};


int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberCounterNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}